addrmgr/addrmanager.go

```diff
@@ -670,6 +670,10 @@ func (a *AddrManager) AddressCache() []*wire.NetAddress {
 	allAddr := make([]*wire.NetAddress, 0, addrIndexLen)
 	// Iteration order is undefined here, but we randomise it anyway.
 	for _, v := range a.addrIndex {
+		// Skip low quality addresses.
+		if v.isBad() {
```

Comment:
I agree with the sentiment here, but I'm fairly certain this will cause a panic later because there is an assumption made below that all addresses end up in the `allAddr` slice.  Notice the code below uses `addrIndexLen` in the calculations which will and up causing `i` and `j` to be out of bounds which would cause a panic.

Reply:
Good catch, sorry for missing that. I adjusted the code accordingly.